### PR TITLE
Mangohud

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@traversaro‎ I could use your knowledge here.

This requires the file 

```
${CONDA_PREFIX}/share/vulkan/implicit_layer.d/MangoHud.x86_64.json
```
to be found at runtime by vulkan

If i symlink it to:
```
sudo ln -s ${CONDA_PREFIX}/share/vulkan/implicit_layer.d/MangoHud.x86_64.json /usr/local/share/vulkan/implicit_layer.d/
```

the overlay seems to showup


```
mangohud vkcube
```
should show an overlay similar to
```
mangohud glxgears
```

but without my hack it does not.

Any ideas?

**Comment by traversaro** [comment]
> @traversaro‎ I could use your knowledge here.
> 
> This requires the file
> 
> ```
> ${CONDA_PREFIX}/share/vulkan/implicit_layer.d/MangoHud.x86_64.json
> ```
> 
> to be found at runtime by vulkan
> 
> If i symlink it to:
> 
> ```
> sudo ln -s ${CONDA_PREFIX}/share/vulkan/implicit_layer.d/MangoHud.x86_64.json /usr/local/share/vulkan/implicit_layer.d/
> ```
> 
> the overlay seems to showup
> 
> ```
> mangohud vkcube
> ```
> 
> should show an overlay similar to
> 
> ```
> mangohud glxgears
> ```
> 
> but without my hack it does not.
> 
> Any ideas?

Definitily not an expert (I did not know what implicit vulkan layers were until 10 minutes ago) but could it make sense to add $CONDA_PREFIX/share to XDG_DATA_DIRS env variable? See https://github.com/KhronosGroup/Vulkan-Loader/blob/1108bba6c97174d172d45470a7470a3d6a564647/docs/LoaderLayerInterface.md#linux-layer-discovery .

**Comment by hmaarrfk** [comment]
> add $CONDA_PREFIX/share to XDG_DATA_DIRS

this did it!

Now should this be done conda-forge wide? seems useful...

**Comment by hmaarrfk** [comment]
I'm going to leave this up until Sept 17 here if anybody from

@conda-forge/vulkan-headers @conda-forge/vulkan-loader @conda-forge/wayland @conda-forge/libglvnd wants to give it a review.

Mangohud is a pretty nifty tool to assess performance of GUIs.

File: recipes/mangohud/build.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    MESON_ARGS="${MESON_ARGS} -Dwith_x11=enabled -Dwith_wayland=enabled -Dwith_dbus=disabled"
+fi
+
+meson setup \
+    build \
+    ${MESON_ARGS} \
+    -Duse_system_spdlog=enabled \
+    -Dinclude_doc=false \
+    -Dwith_nvml=disabled \
+    -Dwith_xnvctrl=disabled \
```
Comment (hmaarrfk) [inline recipes/mangohud/build.sh:13]: hmm without nvctrl you can't seem to see the GPU activity of the nvidia gpu.

**Comment by traversaro** [comment]
> Now should this be done conda-forge wide? seems useful...

Like where? I did this in https://github.com/conda-forge/yarp-feedstock/blob/main/recipe/activate.msh#L3, but I do not think that that feedstock is used a lot. A tricky aspect that the XDG_DATA_DIRS documentation from https://specifications.freedesktop.org/basedir-spec/latest/index.html states:

>  If $XDG_DATA_DIRS is either not set or empty, a value equal to /usr/local/share/:/usr/share/ should be used. 

That means that if there is some piece of software that finds something in `/usr/local/share/` or `/usr/share/` thanks to `XDG_DATA_DIRS` not being set, as soon as we set it the software will break, unless we add `/usr/local/share/:/usr/share/` to XDG_DATA_DIRS as well.

File: recipes/mangohud/conda-forge-xdg-data_dirs.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+     exit 1
+ fi
+ 
++# Implicit layers are hard to find by defining other environment variables
++# We don't define the XDG_DATA_DIRS on conda, so we set it here
++# Since this is a subshell, this won't affect the user's main shell
++# Until we decide that it is worthwhile
++# https://github.com/KhronosGroup/Vulkan-Loader/blob/1108bba6c97174d172d45470a7470a3d6a564647/docs/LoaderLayerInterface.md#linux-layer-discovery
++if [ -z "$XDG_DATA_DIRS" ]; then
++    export XDG_DATA_DIRS="${CONDA_PREFIX}/share"
```
Comment (hmaarrfk) [inline recipes/mangohud/conda-forge-xdg-data_dirs.patch:15]:
```suggestion
+    export XDG_DATA_DIRS="${CONDA_PREFIX}/share:/usr/local/share/:/usr/share/"
```

**Comment by hmaarrfk** [comment]
> Like where? I did this in https://github.com/conda-forge/yarp-feedstock/blob/main/recipe/activate.msh#L3, 

thanks for the pointer. what is `msh` ???

> unless we add /usr/local/share/:/usr/share/ to XDG_DATA_DIRS as well.

thanks my latest build does this.

I found a way to not use activate/deactivate scripts, but I do worry that if this kind of stuff proliferates, many feedstocks will add and remove env vars poorly and will cause more bugs.

**Comment by traversaro** [comment]
> > Like where? I did this in https://github.com/conda-forge/yarp-feedstock/blob/main/recipe/activate.msh#L3,
> 
> thanks for the pointer. what is `msh` ???

https://github.com/mamba-org/multisheller, a project we worked some time ago but that did not found a lot of use, and has some bugs.

**Comment by hmaarrfk** [comment]
ok with nvml working, (tested it on a Nvidia enabled machine) this is now ready for a feedstock. I'll take other inputs there.

Thanks @traversaro

**Pull request merged**